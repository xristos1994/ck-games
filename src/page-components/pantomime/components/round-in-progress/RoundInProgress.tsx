import React, { FC, ReactElement } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { classnames, compose } from '@utils/component-utils';
import { withTranslation, ITranslate } from '@models/i18n/hoc';
import { Button } from '@components';
import { movie, teamNameThatPlaysNow } from '@models/pantomime/props';
import { clockRemainingTime } from '@models/clock/props';
import { setIfMovieFound } from '@models/pantomime/actions';
import { IState } from '@models/interfaces';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require('./styles.module.css');

interface IProps {
  setIfMovieFound: (arg0: boolean) => void;
  movie: string;
  clockRemainingTime: number | null;
  teamNameThatPlaysNow: string | null;
  t: ITranslate;
}

const _RoundInProgress: FC<IProps> = ({
  setIfMovieFound,
  movie,
  clockRemainingTime,
  teamNameThatPlaysNow,
  t
}): ReactElement => {
  return (
    <div className={styles.roundInProgressContainer}>
      <div className={classnames(styles.team)}>{t('Team, it is your turn', [teamNameThatPlaysNow as string])}</div>
      <div className={classnames(styles.movie)}>{movie}</div>
      <div className={classnames(styles.remainingTime)}>{clockRemainingTime}</div>
      <Button onClick={() => setIfMovieFound(true)} className={classnames(styles.movieFoundButton)}>
        {t('Movie Found')}
      </Button>
      <Button onClick={() => setIfMovieFound(false)} className={classnames(styles.movieNotFoundButton)}>
        {t('Movie Not Found')}
      </Button>
    </div>
  );
};

const RoundInProgress = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        movie: IProps['movie'];
        teamNameThatPlaysNow: IProps['teamNameThatPlaysNow'];
        clockRemainingTime: IProps['clockRemainingTime'];
      },
      {
        movie: IProps['movie'];
        teamNameThatPlaysNow: IProps['teamNameThatPlaysNow'];
        clockRemainingTime: IProps['clockRemainingTime'];
      }
    >({
      movie,
      teamNameThatPlaysNow,
      clockRemainingTime
    }),
    { setIfMovieFound }
  ),
  withTranslation
)(_RoundInProgress);

export { RoundInProgress };
