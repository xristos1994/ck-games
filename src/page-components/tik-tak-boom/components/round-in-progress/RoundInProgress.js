import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { classnames } from "@utils/component-utils";
import { Button } from "@components";
import {
  mode,
  syllable,
  playerNameThatPlaysNow,
} from "@models/tik-tak-boom/props";
import styles from "./styles.module.css";
import {
  goToNextPlayer,
  goToPreviousPlayer,
} from "@models/tik-tak-boom/actions";

const _RoundInProgress = ({
  goToNextPlayer,
  goToPreviousPlayer,
  mode,
  syllable,
  playerNameThatPlaysNow,
}) => {
  return (
    <div className={styles.roundInProgressContainer}>
      <div className={classnames(styles.player, "main-color", "largeText")}>
        {playerNameThatPlaysNow} is playing now
      </div>
      <div className={classnames(styles.mode, "main-color", "extraLargeText")}>
        {mode.name}
      </div>
      <div
        className={classnames(
          styles.modeDescription,
          "main-color",
          "normalText"
        )}
      >
        *{mode.description}
      </div>
      <div
        className={classnames(styles.syllable, "main-color", "extraLargeText")}
      >
        {syllable}
      </div>
      <Button
        onClick={() => goToNextPlayer()}
        className={classnames(
          styles.goToNextPlayerButton,
          "primary",
          "extraLargeText"
        )}
      >
        NEXT
      </Button>
      <Button
        onClick={() => goToPreviousPlayer()}
        className={classnames(
          styles.goToPreviousPlayerButton,
          "secondary",
          "largeText"
        )}
      >
        Previous
      </Button>
    </div>
  );
};

const RoundInProgress = connect(
  createStructuredSelector({
    mode,
    syllable,
    playerNameThatPlaysNow,
  }),
  { goToNextPlayer, goToPreviousPlayer }
)(_RoundInProgress);

export { RoundInProgress };
