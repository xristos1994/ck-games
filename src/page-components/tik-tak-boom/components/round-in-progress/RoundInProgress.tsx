import React, { FC, ReactElement } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { classnames, compose } from '@utils/component-utils';
import { withTranslation, ITranslate } from '@models/i18n/hoc';
import { Button } from '@components';
import { mode, syllable, playerNameThatPlaysNow } from '@models/tik-tak-boom/props';
import { goToNextPlayer, goToPreviousPlayer } from '@models/tik-tak-boom/actions';
import { IState } from '@models/interfaces';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const styles = require('./RoundInProgress.module.css');

interface IProps {
  goToNextPlayer: () => void;
  goToPreviousPlayer: () => void;
  mode: { id: string; name: string; description: string };
  syllable: string | null;
  playerNameThatPlaysNow: string | null;
  t: ITranslate;
}

export const _RoundInProgress: FC<IProps> = ({
  goToNextPlayer,
  goToPreviousPlayer,
  mode,
  syllable,
  playerNameThatPlaysNow,
  t
}): ReactElement => {
  return (
    <div className={styles.roundInProgressContainer}>
      <div className={classnames(styles.player)}>
        {t('Player, it is your turn', [playerNameThatPlaysNow as string])}
      </div>
      <div className={classnames(styles.mode)}>{mode && mode.name}</div>
      <div className={classnames(styles.modeDescription)}>*{t(mode.description)}</div>
      <div className={classnames(styles.syllable)}>{syllable}</div>
      <Button onClick={goToNextPlayer} className={classnames(styles.goToNextPlayerButton)}>
        {t('Next Player')}
      </Button>
      <Button onClick={goToPreviousPlayer} className={classnames(styles.goToPreviousPlayerButton)}>
        {t('Previous Player')}
      </Button>
    </div>
  );
};

const RoundInProgress = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        mode: IProps['mode'] | null;
        syllable: IProps['syllable'];
        playerNameThatPlaysNow: IProps['playerNameThatPlaysNow'];
      },
      {
        mode: IProps['mode'] | null;
        syllable: IProps['syllable'];
        playerNameThatPlaysNow: IProps['playerNameThatPlaysNow'];
      }
    >({
      mode,
      syllable,
      playerNameThatPlaysNow
    }),
    { goToNextPlayer, goToPreviousPlayer }
  ),
  withTranslation
)(_RoundInProgress);

export { RoundInProgress };
