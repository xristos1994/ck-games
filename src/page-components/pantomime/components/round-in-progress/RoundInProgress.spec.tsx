import React, { MouseEventHandler, ReactChild } from 'react';
import renderer from 'react-test-renderer';
import { _RoundInProgress as RoundInProgress } from './RoundInProgress';

const Button = (
  { children, onClick, className='' } :
  { children: ReactChild | ReactChild[]; onClick: MouseEventHandler<HTMLButtonElement>; className: string }
) => {
  return (
    <button onClick={onClick} className={className || undefined}>
      {children}
    </button>
  );
};

jest.mock('@components/', () => ({
  __esModule: true,
  Button: Button
}));

const setIfMovieFound = (arg0: boolean) => void arg0;
const movie = 'Movie name';
const clockRemainingTime = 15;
const teamNameThatPlaysNow = 'Team name';
const t = (label: string, param?: string[]) => label + (param ? param[0] : '');

describe('page-components/Pantomime/RoundInProgress', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <RoundInProgress
        setIfMovieFound={setIfMovieFound}
        movie={movie}
        clockRemainingTime={clockRemainingTime}
        teamNameThatPlaysNow={teamNameThatPlaysNow}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
