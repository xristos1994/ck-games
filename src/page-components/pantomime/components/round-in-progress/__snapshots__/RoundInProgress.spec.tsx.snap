// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/Pantomime/RoundInProgress renders correctly 1`] = `
<div
  className="roundInProgressContainer"
>
  <div
    className="team"
  >
    Team, it is your turnTeam name
  </div>
  <div
    className="movie"
  >
    Movie name
  </div>
  <div
    className="remainingTime"
  >
    15
  </div>
  <button
    className="movieFoundButton"
    onClick={[Function]}
  >
    Movie Found
  </button>
  <button
    className="movieNotFoundButton"
    onClick={[Function]}
  >
    Movie Not Found
  </button>
</div>
`;
